runtime/drivers/blob/blobdownloader.go

```diff
@@ -148,138 +169,86 @@ func (it *blobIterator) Close() error {
 
 // HasNext returns true if iterator has more data
 func (it *blobIterator) HasNext() bool {
-	return it.index < len(it.objects)
+	return it.err == nil
 }
 
-// NextBatchSize downloads next n files and copies to local directory
-func (it *blobIterator) NextBatchSize(sizeInBytes int64) ([]string, error) {
+// Next downloads next n files and copies to local directory
+func (it *blobIterator) Next() ([]string, error) {
 	if !it.HasNext() {
-		return nil, io.EOF
+		return nil, it.err
 	}
-	if len(it.nextPaths) != 0 {
+	if len(it.nextPaths) != 0 { // single file path
 		paths := it.nextPaths
-		it.index += len(paths)
 		it.nextPaths = nil
+		it.err = io.EOF
 		return paths, nil
 	}
 
+	if it.downloadedFiles == nil {
+		// start downloading files
+		it.downloadedFiles = make(chan downloadedFile, it.batchCount)
```

Comment:
The impact of a dynamic channel buffer size is really hard to reason about here. Would recommend refactoring to follow the style guide here: https://github.com/uber-go/guide/blob/master/style.md#channel-size-is-one-or-none